### ISSUE TITLE
Enhancement of the handling of unused function parameters in C++

### ISSUE THREAD

**Issue by PlasmaHH** [doxygen/doxygen#10566]
This would probably not be an easy thing to implement, but I thought I might ask anyways, maybe something else comes out of it. 

Extended Context: Sometimes in C++ you have functions with parameters that are not used (e.g. you are overriding a virtual function etc.). Now compilers like gcc have the useful feature of warning you about it, and there are multiple ways to silence them, all with their pros and cons:

* cast the parameter to (void) (clutters the code, especially in single return functions)
* Adding [[maybe_unused]] to the parameter (available since C++17 only, and also looks arguably ugly)
* wrapping the parameter in a macro that evaluates to nothing. doesn't seem ideal, and its a macro
* leaving out the name of the parameter

Now my problem is that ideally I would chose the last one, but that is a bit bad for documentation reasons. Depending on how exactly doxygen is configued and what \copydoc ( or automatic mechanism ) is used, doxygen itself would complain that the parameter is not present and/or undocumented. 

However in some of my projects, the rule is to document every parameter and have doxygen warn/error about that (build is "green" only when there are no doxygen warnings). This precludes using no name, and leaves possibly the macro version as one of the  most attractive as it can for gcc evaluate to nothing and for doxygen just to the parameter name (evaluate only predefined etc.)

Another option came to my mind, in yet some other projects we don't rely too much on doxygen but like having the possibly available argument names visible, and as such code could look like:

```C++
auto foo( int x, bool /*b*/ ) { return x*x; }
```

Note how it almost looks like the parameter is there for those that read only the code.

Despite probably being a bit hard since most likely comments are stripped from the code in an early phase:

TL;DR:

Would it now be possible to have (possibly enabled by some option only ) doxygen recognize this special construct within function parameter lists and use it as the parameter name?

**Comment by PlasmaHH** [comment]
Anyone at least an opinion? I am currently working with an input filter script but thats pretty fragile and error prone since I cannot fully parse C++ there...
